site/tests/universal/homepage.test.ts

```diff
@@ -45,21 +45,23 @@ test("Home page should contain key elements", async ({ page }) => {
     await page.locator('[data-testid="block-slider"] >> .slick-slide').count(),
   ).toBeGreaterThan(5);
 
-  const finalCTA = page.locator('[data-testid="final-cta"]');
+  if (!isMobile) {
+    const finalCTA = page.locator('[data-testid="final-cta"]');
 
-  // @todo: Add tests to handle authenticated users, they shouldn't see this section
-  await expect(finalCTA).toBeVisible();
+    // @todo: Add tests to handle authenticated users, they shouldn't see this section
```

Comment:
I suggest handing authenticated users in another test file. We can keep this one scoped to guests. Too  many `if`s can make tests fragile and complex.

Reply:
I agree.. I intended to write the flow for authenticated users in another `test(...)` in the same file. Happy to move to another test file when I implement